app/services/generate_application_ref.rb

```diff
@@ -0,0 +1,16 @@
+class GenerateApplicationRef
```

Comment:
I think we could generate the application reference when the application form is initially created, instead of having to call the service?